library/public/interactor.h

```diff
@@ -222,6 +222,96 @@ class F3D_EXPORT interactor
   virtual interactor& disableCameraMovement() = 0;
   ///@}
 
+  ///@{ @name Forwarding input events
+  /**
+   * Enumeration of supported mouse buttons.
+   */
+  enum class MouseButton : unsigned char
+  {
+    LEFT,
+    RIGHT,
+    MIDDLE
+  };
+
+  /**
+   * Enumeration of supported mouse wheel directions.
+   */
+  enum class WheelDirection : unsigned char
+  {
+    FORWARD,
+    BACKWARD,
+    LEFT,
+    RIGHT
+  };
+
+  /**
+   * Enumeration of supported input actions.
+   */
+  enum class InputAction : unsigned char
+  {
+    PRESS,
+    RELEASE
+  };
+
+  /**
+   * Enumeration of supported input modifiers.
+   */
+  enum class InputModifier : unsigned char
+  {
+    NONE,
+    CTRL,
+    SHIFT,
+    CTRL_SHIFT
+  };
+
+  /**
+   * Trigger a modifier update.
+   * This will update the internal modifier state of the interactor to match the provided one.
+   */
+  virtual interactor& triggerModUpdate(InputModifier mod) = 0;
+
+  /**
+   * Trigger a mouse button event.
+   * This will trigger the corresponding mouse button press or release event.
+   */
+  virtual interactor& triggerMouseButton(InputAction action, MouseButton button) = 0;
+
+  /**
+   * Trigger a mouse new position event.
+   * Positions are in window coordinates, with (0, 0) being the top-left corner.
+   * The coordinates are expressed in pixels.
+   */
+  virtual interactor& triggerMousePosition(double xpos, double ypos) = 0;
+
+  /**
+   * Trigger a mouse wheel event.
+   * At the moment, only vertical wheel events are observed, but it can change in the future.
+   */
+  virtual interactor& triggerMouseWheel(WheelDirection direction) = 0;
+
+  /**
+   * Trigger a keyboard key event.
+   * This is based on X11 key symbols, it's hard to list all of them, but here are a few:
+   * - "A", "B", "C", ..., "Z" for letters
+   * - "0", "1", "2", ..., "9" for numbers
+   * - "Left", "Right", "Up", "Down" for arrow keys
+   * - "Space", "Return", "Escape", "Tab", "BackSpace" for common keys
+   * - "F1", "F2", ..., "F25" for function keys
+   * - "KP_0", "KP_1", ..., "KP_9" for numpad keys
+   * @note
+   * It's possible to run F3D application in verbose mode and press keys to print their symbols.
```

Comment:
Technically I think it does not need F3D application, should be doable with any libf3d based app.

Reply:
No, because if the libf3d based application relies on the external window (which is likely the case here), pressing the keys won't do anything.

Reply:
good point